### PR TITLE
Collect Docker Hub analytics and GA

### PR DESCRIPTION
**Pull request by the-redback**
> $ analytics run 

This command saves the pull logs in below files,
 
1. https://hub.docker.com/v2/repositories/appscode/?page_size=100 => https://docs.google.com/spreadsheets/d/18lNbYqiP4gsKBoLDoUw2ejOGWN3t3mUOyKGaKeye3kI/

 2. https://hub.docker.com/v2/repositories/kubedb/?page_size=100 => https://docs.google.com/spreadsheets/d/10OGrTJxEDox4VR15U7HPGRjiFpTNfhiUGMuL7BQJKJU/

### CODE REVIEW

**Review by sanjid133 — COMMENTED** [review]
SpreadSheetIdKubeDB and SpreadSheetIdAppsCode  can be read from env variable or file

File: pkg/docker/docker_api.go [hunk 1]
```diff
@@ -0,0 +1,283 @@
+package docker_api
```
Comment (sadlil) [inline pkg/docker/docker_api.go]: package name should match with directory structure. use `package docker`.
- if you want package name as `dockerapi` not `docker_api`

File: pkg/docker/docker_api.go [hunk 2]
```diff
@@ -0,0 +1,283 @@
...
+	"github.com/appscode/errors"
+	"golang.org/x/net/context"
+	"golang.org/x/oauth2/google"
+	"google.golang.org/api/sheets/v4"
+)
+
+type DockerRepositories struct {
+	Count    int         `json:"count"`
+	Next     interface{} `json:"next"`
+	Previous interface{} `json:"previous"`
```
Comment (sadlil) [inline pkg/docker/docker_api.go]: `Next` and `Previous` should *string
As
```json
{
  "count": 93,
  "next": "https://hub.docker.com/v2/repositories/appscode/?page=3&page_size=5",
  "previous": "https://hub.docker.com/v2/repositories/appscode/?page=1&page_size=5"
}

// When there is no page available
{
  "count": 93,
  "next": null,
  "previous": null
}
```

we should get data until `Next = nil`

File: pkg/docker/docker_api.go [hunk 3]
```diff
@@ -0,0 +1,283 @@
...
+	User        string
+	Name        string
+	StarCount   int
+	PullCount   int
+	LastUpdated time.Time
+}
+
+const SpreadSheetIdKubeDB = "10OGrTJxEDox4VR15U7HPGRjiFpTNfhiUGMuL7BQJKJU"   //https://docs.google.com/spreadsheets/d/10OGrTJxEDox4VR15U7HPGRjiFpTNfhiUGMuL7BQJKJU
+const SpreadSheetIdAppsCode = "18lNbYqiP4gsKBoLDoUw2ejOGWN3t3mUOyKGaKeye3kI" //https://docs.google.com/spreadsheets/d/18lNbYqiP4gsKBoLDoUw2ejOGWN3t3mUOyKGaKeye3kI
+
```
Comment (sadlil) [inline pkg/docker/docker_api.go]: If we are going to stay with google docs now, at least make them as flags or EnvVar

File: pkg/docker/docker_api.go [hunk 4]
```diff
@@ -0,0 +1,283 @@
...
+func getDockerLogs(link string) (*DockerRepositories, error) {
+
```
Comment (sadlil) [inline pkg/docker/docker_api.go]: why those blank lines. remove unnecessary blank lines.

File: pkg/docker/docker_api.go [hunk 5]
```diff
@@ -0,0 +1,283 @@
...
+	url := link
+
+	// Build the request
+	req, err := http.NewRequest("GET", url, nil)
+	if err != nil {
```
Comment (sadlil) [inline pkg/docker/docker_api.go]: You should be careful about `Next` page of response.

File: pkg/docker/docker_api.go [hunk 6]
```diff
@@ -0,0 +1,283 @@
...
+
+	return &record, nil
+}
+
+func updateSheet(dockLogs DockerRepoLogs, SpreadSheetId string) (string, error) {
+	// Usase Limits: https://developers.google.com/sheets/api/limits
+	// 100 Requeste per 100Seconds per User
+	// 500 requests per 100 seconds per project
+	// So, making it slow each request
+	time.Sleep(5 * time.Second)
```
Comment (sadlil) [inline pkg/docker/docker_api.go]: time.Sleep doesn't make it slow.
if you are careful about rate limit, then you should call this method when ever it is ok to call.
like call this method every 5 seconds and update data.

File: pkg/docker/docker_api.go [hunk 7]
```diff
@@ -0,0 +1,283 @@
...
+	config, err := google.ConfigFromJSON(b, "https://www.googleapis.com/auth/spreadsheets")
+	if err != nil {
+		log.Println("Unable to parse client secret file to config")
+	}
+	client := spreadsheet.GetClient(ctx, config)
+	srv, err := sheets.New(client)
+	if err != nil {
+		log.Println("Unable to read client secret file: ")
+	}
+
```
Comment (sadlil) [inline pkg/docker/docker_api.go]: remove the previous repetation. move them to a separate method or function and use them as where necessary.

**Review by sadlil — CHANGES_REQUESTED** [review]

File: pkg/cmds/server.go [hunk 8]
```diff
@@ -26,6 +28,10 @@ func NewCmdServer(version string) *cobra.Command {
 			analytics.SendEvent("analytics", "stopped", version)
 		},
 		Run: func(cmd *cobra.Command, args []string) {
+			c := cron.New()
+			c.AddFunc("@every 4h", dockerapi.DockerAnalytics)
```
Comment (tamalsaha) [inline pkg/cmds/server.go]:
```
c.AddFunc("@every 4h", func(){
	if err := dockerapi.DockerAnalytics(); err != nil {
		log.Errorln(err)
	}
})
```

**Pull request merged**